packages/flutter_tools/lib/src/build_system/targets/linux.dart

```diff
@@ -44,22 +51,50 @@ class UnpackLinuxDebug extends Target {
   @override
   Future<void> build(Environment environment) async {
     final String basePath = artifacts.getArtifactPath(Artifact.linuxDesktopPath);
-    for (File input in fs.directory(basePath)
-        .listSync(recursive: true)
-        .whereType<File>()) {
-      final String outputPath = fs.path.join(
-        environment.projectDir.path,
-        'linux',
-        'flutter',
-        'ephemeral',
-        fs.path.relative(input.path, from: basePath),
-      );
-      final File destinationFile = fs.file(outputPath);
-      if (!destinationFile.parent.existsSync()) {
-        destinationFile.parent.createSync(recursive: true);
+    final List<File> inputs = <File>[];
+    final List<File> outputs = <File>[];
+    final String outputPrefix = fs.path.join(
+      environment.projectDir.path,
+      'linux',
+      'flutter',
+      'ephemeral',
+    );
+    for (String artifact in _kLinuxArtifacts) {
+      final String entityPath = fs.path.join(basePath, artifact);
+      if (fs.isFileSync(entityPath)) {
+        final String outputPath = fs.path.join(
+          outputPrefix,
+          fs.path.relative(entityPath, from: basePath),
+        );
+        final File destinationFile = fs.file(outputPath);
+        if (!destinationFile.parent.existsSync()) {
+          destinationFile.parent.createSync(recursive: true);
+        }
+        final File inputFile = fs.file(entityPath);
+        inputFile.copySync(destinationFile.path);
+        inputs.add(inputFile);
+        outputs.add(destinationFile);
+        continue;
```

Comment:
nit: this code might need some comments.

Reply:
I tried to add some more context here, PTAL